### PR TITLE
ZEN-33016 New zenhubworker service for flexible per-customer tuning

### PR DESCRIPTION
**Pull request by romanbakaleyko**

### CODE REVIEW

File: Products/ZenHub/server/config.py [hunk 1]
```diff
@@ -59,6 +61,34 @@ def xmlrpcport(self):
...
+    def __init__(self, config):
+
+        self.__config = {}
+
+        try:
+            with open(config, 'r') as f:
+                self.__config = json.load(f)
+        except Exception as e:
+            import logging
+            log = logging.getLogger("zen.%s" %  __name__)
```
Comment (jpeacock-zenoss) [inline Products/ZenHub/server/config.py:76]: Please name this logger "zen.zenhub.server.config" to stay consistent with the other loggers defined in zenhub, e.g.
```
    log = logging.getLogger("zen.zenhub.server.config")
```
The `__name__` variable will return "Products.ZenHub.server.config" which is not the desired name.

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]
I think the configuration file for specifying routes is too complicated.  Configuration files are just as much an API of a service as methods are for a class.

I have written an alternate configuration file for configuring the executors, pools, and routes.  It groups details about a particular executor together in one place.  The alternative format also removes redundant information. I opened a PR on this PR here: https://github.com/zenoss/zenoss-prodbin/pull/3982.

Please take a look.

**Review by jpeacock-zenoss — CHANGES_REQUESTED** [review]
There should be unit tests to cover this new code.

File: Products/ZenHub/server/config.py (hunk 1 above)
Comment (romanbakaleyko) [reply Products/ZenHub/server/config.py]: done

**Review by jpeacock-zenoss — APPROVED** [review]

**Pull request merged**